src/cc_toolchain/cc_toolchain_config.bzl

```diff
@@ -0,0 +1,791 @@
+        ],
+    )
+
+    lld_feature = feature(
+        name = "lld",
+        flag_sets = [
+            flag_set(
+                actions = ALL_LINK_ACTIONS,
+                flag_groups = [flag_group(flags = [
+                    "-fuse-ld=lld",
+                    "-rtlib=compiler-rt",
+                ])],
+            ),
+        ],
+    )
+
+    opt_feature = feature(
+        name = "opt",
+        flag_sets = [
+            flag_set(
+                actions = [ACTION_NAMES.c_compile, ACTION_NAMES.cpp_compile],
+                flag_groups = [
+                    flag_group(
+                        flags = [
+                            "-ggdb",
+                            "-O3",
+                            "-ffunction-sections",
+                            "-fdata-sections",
+                        ],
+                    ),
+                ],
+            ),
+            flag_set(
+                actions = [
+                    ACTION_NAMES.cpp_link_dynamic_library,
+                    ACTION_NAMES.cpp_link_nodeps_dynamic_library,
+                    ACTION_NAMES.cpp_link_executable,
+                ],
+                flag_groups = [flag_group(flags = ["-Wl,--gc-sections"])],
+            ),
+        ],
+        implies = ["common"],
+    )
+
+    clang_warnings_feature = feature(
+        name = "clang_warnings_feature",
+        flag_sets = [
+            flag_set(
+                actions = [ACTION_NAMES.c_compile, ACTION_NAMES.cpp_compile],
+                flag_groups = [
+                    flag_group(
+                        flags = ["-Wno-invalid-constexpr"],
+                    ),
+                ],
+            ),
+        ],
+        implies = ["common"],
+    )
+
+    common_feature = feature(
+        name = "common",
+        implies = [
+            "stdlib",
+            "c++17",
+            "determinism",
+            "hardening",
+            "clang_warnings_feature",
+            "build-id",
+            "no-canonical-prefixes",
+            "lld",
+            "frame-pointer",
+            "static_link_cpp_runtimes",
+        ],
+    )
+
+    features = common.values() + [
+        static_libgcc,
+        pic_feature,
+        supports_pic_feature,
+        stdlib_feature,
+        common_feature,
+        lld_feature,
+        opt_feature,
+        runtime_library_search_directories,
+        clang_warnings_feature,
+    ]
+
+    cxx_builtin_include_directories = ctx.attr.builtin_include_directories
+
+    tool_paths = [
+        tool_path(name = "gcc", path = ctx.attr.host_compiler_path),
+        tool_path(name = "ar", path = ctx.attr.host_compiler_prefix + "/clang-ar"),
+        tool_path(name = "compat-ld", path = ctx.attr.host_compiler_prefix + "/clang-lld"),
+        tool_path(name = "cpp", path = ctx.attr.host_compiler_prefix + "/clang-cpp"),
+        tool_path(name = "dwp", path = ctx.attr.host_compiler_prefix + "/clang-dwp"),
+        tool_path(name = "gcov", path = "fake"),
```

Comment:
What does this fake mean?

Reply:
I'm not sure. I've also just noticed that coverage did not get uploaded for this PR, so gonna have to look at that.........

Reply:
Maybe I should've asked this before, but why are we moving to clang?

Reply:
For a few reasons:
- compile times should be a bit better
- it was the best toolchain config I could find (I didn't write this all myself). 
- If you're not using a toolchain bazel just tries to be smart and guess that to use based on your system. This has the added benefit of making sure everything is using exactly the same compiler.

Reply:
Cool! So we need a specific toolchain for fw and that's why everything needs to move over?

Reply:
We could probably have gotten away with using whatever bazel guessed from our system for non-firmware builds, but it would've been a bit hacky and the toolchain config came with `clang` anyhow.